shared/src/types/key/secp256k1.rs

```diff
@@ -405,12 +435,19 @@ impl PartialOrd for Signature {
     }
 }
 
-impl TryFrom<&[u8; 64]> for Signature {
+impl TryFrom<&[u8; 65]> for Signature {
     type Error = ParseSignatureError;
 
-    fn try_from(sig: &[u8; 64]) -> Result<Self, Self::Error> {
-        libsecp256k1::Signature::parse_standard(sig)
-            .map(Self)
+    fn try_from(sig: &[u8; 65]) -> Result<Self, Self::Error> {
+        let sig_bytes = sig[..64].try_into().unwrap();
+        let recovery_id = RecoveryId::parse(sig[64]).map_err(|err| {
```

Comment:
We could use `SIGNATURE_LENGTH`/`SIGNATURE_LENGTH - 1` here as well instead of `65`/`64`